src/chainhooks/types.rs

```diff
@@ -0,0 +1,354 @@
+        } else {
+            return Err(format!(
+                "chain '{}' not supported (stacks, bitcoin)",
+                self.chain
+            ));
+        };
+        Ok(res)
+    }
+
+    pub fn to_bitcoin_specification(
+        &self,
+        network: &BitcoinNetwork,
+    ) -> Result<BitcoinChainhookSpecification, String> {
+        let network_ser = format!("{:?}", network).to_lowercase();
+        let network_spec = match self.networks.get(&network_ser) {
+            Some(entry) => entry,
+            None => {
+                return Err(format!(
+                    "network '{}' not found in chainhook specification file",
+                    network_ser
+                ))
+            }
+        };
+
+        Ok(BitcoinChainhookSpecification {
+            uuid: format!("{}", self.id.unwrap_or(1)),
+            version: self.version.unwrap_or(1),
+            name: self.name.to_string(),
+            network: network.clone(),
+            start_block: network_spec.start_block,
+            end_block: network_spec.end_block,
+            predicate: network_spec.predicate.to_bitcoin_predicate()?,
+            action: network_spec.action.to_specifications()?,
+        })
+    }
+
+    pub fn to_stacks_specification(
+        &self,
+        network: &StacksNetwork,
+    ) -> Result<StacksChainhookSpecification, String> {
+        let network_ser = format!("{:?}", network).to_lowercase();
+        let network_spec = match self.networks.get(&network_ser) {
+            Some(entry) => entry,
+            None => {
+                return Err(format!(
+                    "network '{}' not found in chainhook specification file",
+                    network_ser
+                ))
+            }
+        };
+
+        Ok(StacksChainhookSpecification {
+            uuid: format!("{}", self.id.unwrap_or(1)),
+            version: self.version.unwrap_or(1),
+            name: self.name.to_string(),
+            network: network.clone(),
+            start_block: network_spec.start_block,
+            end_block: network_spec.end_block,
+            predicate: network_spec.predicate.to_stacks_predicate()?,
+            action: network_spec.action.to_specifications()?,
+        })
+    }
+}
+
+impl HookActionFile {
+    pub fn to_specifications(&self) -> Result<HookAction, String> {
+        if let Some(ref specs) = self.http {
+            let url = match specs.get("url") {
+                Some(url) => Ok(url.to_string()),
+                None => Err(format!("url missing for http")),
+            }?;
+            let method = match specs.get("method") {
+                Some(method) => Ok(method.to_string()),
+                None => Err(format!("method missing for http")),
+            }?;
+            let authorization_header = match specs.get("authorization-header") {
+                Some(authorization_header) => Ok(authorization_header.to_string()),
+                None => Err(format!("method missing for http")),
```

Comment:
```suggestion
                None => Err(format!("authorization-header missing for http")),
```

Reply:
Addressed in https://github.com/hirosystems/clarinet/pull/402/commits/ce821313ed66b9190826dd147cdcc477cf6908f0